lib/shared/sudo/linux.ts

```diff
@@ -0,0 +1,142 @@
+/* 
+ * This is heavily inspired (read: a ripof) https://github.com/balena-io-modules/sudo-prompt
+ * Which was a fork of https://github.com/jorangreef/sudo-prompt
+ * 
+ * This and the original code was released under The MIT License (MIT)
+ * 
+ * Copyright (c) 2015 Joran Dirk Greef
+ * Copyright (c) 2024 Balena
+ * 
+    The MIT License (MIT)
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+ */
+
+import { spawn } from 'child_process';
+import { access, constants } from 'fs/promises';
+import { env } from 'process';
+
+// const execFileAsync = promisify(execFile);
+
+const SUCCESSFUL_AUTH_MARKER = 'AUTHENTICATION SUCCEEDED';
+
+/** Check for kdesudo or pkexec */
+function checkLinuxBinary() {
+	// eslint-disable-next-line no-async-promise-executor
+	return new Promise(async (resolve, reject) => {
+		// We used to prefer gksudo over pkexec since it enabled a better prompt.
+		// However, gksudo cannot run multiple commands concurrently.
+
+		const paths = ['/usr/bin/kdesudo', '/usr/bin/pkexec'];
+		for (const path of paths) {
+			try {
+				// check if the file exist and is executable
+				await access(path, constants.X_OK);
+				resolve(path);
+			} catch (error: any) {
+				continue;
+			}
+		}
+		reject('Unable to find pkexec or kdesudo.');
+	});
+}
+
+function EscapeDoubleQuotes(escapeString: string) {
```

Comment:
What’s with the non-camelCase names in this PR? Why the need to diverge?

Reply:
that one is probably because I copypasted the function from the original package and didn't realized it wasn't camel